core/src/main/java/org/apache/iceberg/util/ManifestFileUtil.java

```diff
@@ -59,16 +59,16 @@ boolean canContain(Object value) {
         return containsNull;
       }
 
-      if (NaNUtil.isNaN(value)) {
-        return containsNaN;
-      }
-
       // if lower bound is null, then there are no non-null values
       if (lowerBound == null) {
         // the value is non-null, so it cannot match
         return false;
```

Comment:
I think if the input is NaN and the file is written with the new code (`NaN` flips `containsNaN` to true without touching upper/lower bound) then I think we will return false here which wouldn't be correct?

Reply:
I think @yyanyy is correct. It was my bad for suggesting it.

Reply:
Ok I think i understand the issue here,

Previously we would report "null" for the lowerBound when we meant "NaN" because we weren't recording NaNs, this means if we see a missing lowerBound we either could have a mix of NaN and Null

Reply:
Fixed !